var/spack/repos/builtin/packages/gaudi/package.py

```diff
@@ -63,34 +91,27 @@ class Gaudi(CMakePackage):
     depends_on("python", type=("build", "run"))
     depends_on("python@:3.7", when="@32.2:34", type=("build", "run"))
     depends_on("python@:2", when="@:32.1", type=("build", "run"))
+    depends_on("py-networkx@:2.2", when="^python@:2.7")
+    depends_on("py-networkx", when="^python@3.0.0:")
+    depends_on("py-nose", type="test")
     depends_on("py-setuptools@:45", when="^python@:2.7", type="build")
     depends_on("py-six", type=("build", "run"))
     depends_on("py-xenv@1:", when="@:34.9", type=("build", "run"))
     depends_on("range-v3")
     depends_on("root +python +root7 +ssl +tbb +threads")
     depends_on("zlib")
 
-    # todo: this should be a test dependency only,
-    depends_on("py-nose", when="@35.0:36.1", type=("build", "run"))
-    depends_on("py-pytest", when="@36.2:", type=("build", "run"))
-
     # Adding these dependencies triggers the build of most optional components
-    depends_on("cppgsl", when="+optional")
-    depends_on("cppunit", when="+optional")
+    depends_on("cppgsl", when="+cppunit")
+    depends_on("cppunit", when="+cppunit")
     depends_on("doxygen +graphviz", when="+docs")
-    depends_on("gperftools", when="+optional")
-    depends_on("gdb", when="+optional")
-    depends_on("gsl", when="+optional")
-    depends_on("heppdt@:2", when="+optional")
-    depends_on("jemalloc", when="+optional")
-    depends_on("libpng", when="+optional")
-    depends_on("libunwind", when="+optional")
-    depends_on("py-networkx@:2.2", when="+optional ^python@:2.7")
-    depends_on("py-networkx", when="+optional ^python@3.0.0:")
-    depends_on("py-setuptools", when="+optional")
-    depends_on("py-nose", when="+optional")
-    depends_on("relax", when="@:33 +optional")
-    depends_on("xerces-c", when="+optional")
+    depends_on("gperftools", when="+gperftools")
+    depends_on("gdb")
+    depends_on("heppdt", when="+heppdt")
+    depends_on("jemalloc", when="+jemalloc")
+    depends_on("libunwind", when="+unwind")
+    # depends_on("relax", when="@:33 +optional")
```

Comment:
RELAX is required through v34r1 for examples, https://gitlab.cern.ch/gaudi/Gaudi/-/blob/v34r1/GaudiExamples/CMakeLists.txt#L26.
```suggestion
    depends_on("relax", when="@:34 +examples")
```